soweego/target_selection/musicbrainz/musicbrainz_baseline_matcher.py

```diff
@@ -66,39 +53,70 @@ def equal_strings_match():
         urls = csv.DictReader(tsvfile, dialect='excel-tab', fieldnames=fieldnames)
         for url in urls:
             domain = re.search(domain_regex, url[2]).group(1)
             domains[domain] += 1
     towrite = {domain:count for (domain, count) in domains.items() if count > 10000}
-    json.dump(towrite, open('%s/urls.json' % get_output_path(), 'w'), indent=2, ensure_ascii=False)
+    json.dump(towrite, open('%s/urls.json' % common.get_output_path(), 'w'), indent=2, ensure_ascii=False)
 
-def get_users_urls():
-    """Creates the json containing url - artist id"""
+def _get_users_urls(dump_folder_path, output):
+    output_full_path = os.path.join(output, 'url_artist.json')
     urlid_id = defaultdict(str)
     url_id = defaultdict(str)
-    
-    with open('%s/musicbrainz_dump_20180725-001823/mbdump/l_artist_url' % get_path(), "r") as tsvfile:
-        fieldnames = [i for i in range(0, 6)]
-        url_relationships = csv.DictReader(tsvfile, dialect='excel-tab', fieldnames=fieldnames)
-        for relationship in url_relationships:
-            # url id matched with its user id
-            urlid_id[relationship[3]] = relationship[2]
-
-    with open('%s/musicbrainz_dump_20180725-001823/mbdump/url' % get_path(), "r") as tsvfile:
-        fieldnames = [i for i in range(0, 5)]
-        urls = csv.DictReader(tsvfile, dialect='excel-tab', fieldnames=fieldnames)
-        for url in urls:
-            # Translates the url ids stored before in the respective urls
-            if url[0] in urlid_id:
-                url_id[url[2]] = urlid_id[url[0]]
 
-    json.dump(url_id, open('%s/url_artist.json' % get_output_path(), 'w'), indent=2, ensure_ascii=False)    
+    if os.path.isfile(output_full_path):
+        return json.load(open(output_full_path))
+    else:
+        with open(os.path.join(dump_folder_path, 'mbdump/l_artist_url'), "r") as tsvfile:
+            fieldnames = [i for i in range(0, 6)]
+            url_relationships = csv.DictReader(tsvfile, dialect='excel-tab', fieldnames=fieldnames)
+            for relationship in url_relationships:
+                # url id matched with its user id
+                urlid_id[relationship[3]] = relationship[2]
+
+        with open(os.path.join(dump_folder_path, 'mbdump/url'), "r") as tsvfile:
+            fieldnames = [i for i in range(0, 5)]
+            urls = csv.DictReader(tsvfile, dialect='excel-tab', fieldnames=fieldnames)
+            for url in urls:
+                # Translates the url ids stored before in the respective urls
+                if url[0] in urlid_id:
+                    url_id[url[2]] = urlid_id[url[0]]
+
+        json.dump(url_id, open(output_full_path, 'w'), indent=2, ensure_ascii=False)
+        return url_id
+
+@click.command()
+@click.argument('dump_folder_path', type=click.Path(exists=True))
+@click.option('--output', '-o', default=common.get_output_path(), type=click.Path(exists=True))
```

Comment:
Do not use `common.get_output_path()`, as it would create a directory in the package location. We are calling the CLI from the root directory, so the output folder would be hidden in some subdirectory.
Just use `'output'` as default value.